resources/airflow.rb

```diff
@@ -1306,10 +1306,17 @@ def self.create_ducts_objects(model, runner, building, ducts, mech_vent, tin_sen
 
         if not leakage_fracs[Constants.DuctSideSupply].nil?
           duct_subroutine.addLine("  Set f_sup = #{leakage_fracs[Constants.DuctSideSupply]}") # frac
-          duct_subroutine.addLine("  Set f_ret = #{leakage_fracs[Constants.DuctSideReturn]}") # frac
         elsif not leakage_cfm25s[Constants.DuctSideSupply].nil?
           duct_subroutine.addLine("  Set f_sup = #{UnitConversions.convert(leakage_cfm25s[Constants.DuctSideSupply], "cfm", "m^3/s").round(6)} / (#{fan_mfr_max_var.name} * 1.0135)") # frac
+        else
+          duct_subroutine.addLine("  Set f_sup = 0.0") # frac
+        end
+        if not leakage_fracs[Constants.DuctSideReturn].nil?
+          duct_subroutine.addLine("  Set f_ret = #{leakage_fracs[Constants.DuctSideReturn]}") # frac
+        elsif not leakage_cfm25s[Constants.DuctSideReturn].nil?
           duct_subroutine.addLine("  Set f_ret = #{UnitConversions.convert(leakage_cfm25s[Constants.DuctSideReturn], "cfm", "m^3/s").round(6)} / (#{fan_mfr_max_var.name} * 1.0135)") # frac
+        else
+          duct_subroutine.addLine("  Set f_ret = 0.0") # frac
```

Comment:
The code previously assumed there was both supply and return leakage in each duct location zone. Now it sets `f_sup` (or `f_ret`) to zero if there's no supply (or return) leakage.

Reply:
👍